lib/modules/manager/composer/artifacts.ts

```diff
@@ -25,21 +25,33 @@ import type { AuthJson, ComposerLock } from './types';
 import {
   composerVersioningId,
   extractConstraints,
+  findGithubPersonalAccessToken,
   getComposerArguments,
   getPhpConstraint,
   requireComposerDependencyInstallation,
 } from './utils';
+import { GitTagsDatasource } from '../../datasource/git-tags';
 
 function getAuthJson(): string | null {
   const authJson: AuthJson = {};
 
-  const githubCredentials = hostRules.find({
+  const githubToken = findGithubPersonalAccessToken({
     hostType: PlatformId.Github,
     url: 'https://api.github.com/',
   });
-  if (githubCredentials?.token) {
+  if (githubToken) {
     authJson['github-oauth'] = {
-      'github.com': githubCredentials.token.replace('x-access-token:', ''),
+      'github.com': githubToken,
+    };
+  }
+
+  const gitTagsGithubToken = findGithubPersonalAccessToken({
+    hostType: GitTagsDatasource.id,
+    url: 'https://github.com',
+  });
+  if (gitTagsGithubToken) {
+    authJson['github-oauth'] = {
+      'github.com': gitTagsGithubToken,
     };
   }
```

Comment:
Is it intended that a git-tags token will override a github token?

Reply:
Yes, this is the whole point of the pull request